### ISSUE TITLE
Can't insert power source data into database for GT06 device.

### ISSUE THREAD

**Issue by zizango123** [traccar/traccar#1640]
Hi, I have my own traccar server setup and am getting co-ordinate, speed datas. But I am not able to get the power source used (i.e vehicle's power source or internal tracker battery ) . How can i insert this data into the database.

Note: i know it uses the config file for sql insertion but in cannot seem to get the sql format for this task.

**Comment by tananaev** [comment]
What version of Traccar are you using? Are you sure your device sends power/battery information?

**Comment by zizango123** [comment]
i use the official latest traccar and the device that i use is the concox gt06

**Comment by tananaev** [comment]
GT06 protocol includes two main types of messages:

1. GPS info which doesn't include power info
2. Status information including battery/power

I guess your device sends only the first type.

**Comment by zizango123** [comment]
i have checked and it does send both of the messages . The only problem is how to retrieve it

**Comment by tananaev** [comment]
All additional data (e.g. battery etc) is saved in `attributes` field in the `position` database table.

**Comment by ysvseven** [comment]
@tananaev 
Hi Anton , Ive noticed that most Concox (from JIMI IOT or Concox) devices send the external battery paramater in Status information packet not in (GPS info) and Tracacr parse it as adc1 
e.g message - "79790008940004d8002baed10d0a" contains external voltage (00 04 d8 00)
can we add this to GT06 protocol to parse as "Power" ?

**Comment by tananaev** [comment]
Are you sure it's not decoded? Do you have documentation?

**Comment by ysvseven** [comment]
Hi Anton ,
Yes not its not decoded , I tested with a few devices .
Attaching the protocol document.


[Protocol(JM-VL02,JM-VG03,JM-VG04,EG02,EG03,JM01,JV200,GT300,GT800,MT200,OB22,X3,Q2,GT08,Wetrack lite,ET25,HVT001）V3.2 20240222.pdf](https://github.com/user-attachments/files/24324575/Protocol.JM-VL02.JM-VG03.JM-VG04.EG02.EG03.JM01.JV200.GT300.GT800.MT200.OB22.X3.Q2.GT08.Wetrack.lite.ET25.HVT001.V3.2.20240222.pdf)

**Comment by tananaev** [comment]
Is that sample from your logs?

**Comment by ysvseven** [comment]
Yes ,
And adding some logs here as well 
79790008940004d8002baed10d0a 
79790008940004d10053cd000d0a 
79790008940000000078f5ee0d0a

**Comment by tananaev** [comment]
I checked and it is decoded as ADC.

**Comment by ysvseven** [comment]
Yes you are correct , it decos as adc1 , what I would like to request is to decode it as Power .
So its not required to make computed Atribute to make adc1 as Power
Will it be possible ?

**Comment by tananaev** [comment]
No because some models define it as ADC, not external power.

**Comment by tananaev** [comment]
No because some models define it as ADC, not external power.

**Comment by ysvseven** [comment]
OK thanks .
